Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/create/RATProfileCreateFragmentViewModel.kt

```diff
@@ -6,31 +6,40 @@ import androidx.lifecycle.asLiveData
 import dagger.assisted.Assisted
 import dagger.assisted.AssistedFactory
 import dagger.assisted.AssistedInject
-import de.rki.coronawarnapp.coronatest.antigen.profile.RATProfileSettingsDataStore
+import de.rki.coronawarnapp.profile.model.Profile
+import de.rki.coronawarnapp.profile.storage.ProfileRepository
 import de.rki.coronawarnapp.util.ui.SingleLiveEvent
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
+import kotlinx.coroutines.flow.map
 import org.joda.time.LocalDate
 import org.joda.time.format.DateTimeFormat
 import org.joda.time.format.DateTimeFormatter
 import timber.log.Timber
 
 class RATProfileCreateFragmentViewModel @AssistedInject constructor(
-    private val ratProfileSettings: RATProfileSettingsDataStore,
+    private val profileRepository: ProfileRepository,
     @Assisted private val format: DateTimeFormatter = DateTimeFormat.mediumDate()
 ) : CWAViewModel() {
 
-    private val profileData = MutableLiveData(RATProfileData())
-    val profile: LiveData<RATProfileData> = profileData
+    // TO DO get id as nav arg
+    private val id: String? = "1"
```

Comment:
it should not be optional specially  when needed in navigation

Reply:
this is for creating new profiles, then the id is null